src/test/java/org/cactoos/iterator/MatchedTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterator;
+
+import java.util.Objects;
+import org.cactoos.list.ListOf;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.HasValues;
+import org.llorllale.cactoos.matchers.Throws;
+
+/**
+ * Test case for {@link Mapped}.
+ * @since 0.47
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+final class MatchedTest {
+    @Test
+    void failsWhenElementsNotMatch() {
+        new Assertion<>(
+            "All elements have correlation function as `equal`",
+            () -> new ListOf<>(
+                new Matched<>(
+                    Objects::equals,
+                    new IteratorOf<>(1),
+                    new IteratorOf<>(0)
+                )
+            ),
+            new Throws<>(
+                new IsEqual<>(
+                    "The is no correlation between `1` and `0`."
+                ),
+                IllegalStateException.class
+            )
+        ).affirm();
+    }
+
+    @Test
+    void failsOnSizeMismatch() {
+        new Assertion<>(
+            "must fail if sizes are different",
```

Comment:
@andreoss Please, be more specific: `sizes are different`.... sizes of what?

Reply:
@fabriciofx Done.